components/frontend/src/AppUI.js

```diff
@@ -13,6 +13,22 @@ import { DarkMode } from './context/DarkMode';
 import { Permissions } from './context/Permissions';
 import { PageContent } from './PageContent';
 import { getUserPermissions, userPrefersDarkMode, useURLSearchQuery } from './utils'
+import { get_settings, post_settings } from './api/settings';
+
+const DEFAULT_SETTINGS = {
+    date_interval: 7,
+    date_order: "descending",
+    hidden_columns: [],
+    hide_metrics_not_requiring_action: false,
+    nr_dates: 1,
+    sort_column: null,
+    sort_direction: "ascending",
+    tabs: [],
+    show_issue_summary: false,
+    show_issue_creation_date: false,
+    show_issue_update_date: false,
+    ui_mode: null
+}
```

Comment:
I think the introduction of the default settings (good idea) could well be done as a separate PR. Makes this one smaller, and makes it easier to get high test coverage.